cmd/thv-operator/controllers/mcpserver_controller.go

```diff
@@ -621,6 +623,36 @@ func (r *MCPServerReconciler) deploymentForMCPServer(m *mcpv1alpha1.MCPServer) *
 	return dep
 }
 
+func ensureRequiredEnvVars(env []corev1.EnvVar) []corev1.EnvVar {
+	// Check for the existence of the XDG_CONFIG_HOME and HOME environment variables
+	// and set them to /tmp if they don't exist
+	xdgConfigHomeFound := false
```

Comment:
@ChrisJBurns do we have a ticket to ensure the proxy runner doesn't write any files? I think in general the proxy runner should be able to run with read-only filesystem

Reply:
Ooops almost missed this! Are you speaking about the general `securityContext` for the ProxyRunner (`readOnlyFileSystem: true`) or specifically about the code itself? I think for Kubernetes, it doesn't write to any files, and even if it did, I can't think of any reason why they would be used. Mainly because the Client configurations only matter for local setups. However, that being said, I don't see us setting the [`securityContext`](https://github.com/stacklok/toolhive/blob/main/cmd/thv-operator/controllers/mcpserver_controller.go#L562) in the code, so I can create a ticket to do some best-practice stuff there like we do for the actual MCPServer itself.

Reply:
Created https://github.com/stacklok/toolhive/issues/1285 to track the ProxyRunner secContext